src/backend/optimizer/util/plancat.c

```diff
@@ -510,7 +518,40 @@ cdb_estimate_rel_size(RelOptInfo   *relOptInfo,
 		 * to derive curpages, else use the default value.
 		 */
 		if (gp_enable_relsize_collection)
-			curpages = cdbRelMaxSegSize(rel) / BLCKSZ;
+		{
+			if (RelationIsAppendOptimized(rel))
+			{
+				FileSegTotals *totals;
+				if (RelationIsAoRows(rel))
+					totals = GetSegFilesTotalsCluster(rel);
+				else
+					totals = GetAOCSSegFilesTotalsCluster(rel);
+
+				Assert(totals);
+
+				curpages = totals->totalbytesuncompressed / BLCKSZ;
```

Comment:
Why is `curpages` calculated based on `totalbytesuncompressed`, but not `totalbytes`? Am I right that the `pages` output argument is used to estimated disk I/O?